Classes/ViewHelpers/FormViewHelper.php

```diff
@@ -58,10 +67,15 @@ protected function getFormObjectName()
     protected function getFormActionUri()
     {
         /** @var ActionRequest $actionRequest */
-        $actionRequest = $this->controllerContext->getRequest();
-        $uri = $actionRequest->getHttpRequest()->getUri();
-        if ($this->hasArgument('section') && $this->arguments['section'] !== '') {
-            $uri = $uri->withFragment($this->arguments['section']);
+        $actionRequest = clone $this->controllerContext->getRequest();
+        $requestUri = $actionRequest->getHttpRequest()->getUri();
+        $uri = $this->baseUriProvider->getConfiguredBaseUriOrFallbackToCurrentRequest()
+            ->withPath($requestUri->getPath())
+            ->withQuery($requestUri->getQuery())
+            ->withFragment($requestUri->getFragment());
+
+        if ($this->hasArgument('section')) {
+            $uri = preg_replace('/#.*$/', '', $uri) . '#' . $this->arguments['section'];
```

Comment:
Wouldn't `$uri = $uri->withFragment($this->arguments['section']);` make more sense/be more readable? Or is there a specific reason to do this manually like that?

Reply:
yes, I aggree
```suggestion
            $uri = $uri->withFragment($this->arguments['section']);
```